game-core/src/main/java/games/strategy/engine/data/IAttachment.java

```diff
@@ -23,4 +25,6 @@
   String getName();
 
   void setName(String name);
+
+  Map<String, Function<IAttachment, AttachmentProperty<?>>> getAttachmentMap();
```

Comment:
This I think is a pretty good candidate for a 1st class type:
``` Map<String, Function<IAttachment, AttachmentProperty<?>>>```

Eg:
```
AttachmentMap getAttachmentMap();
```

Reply:
Also, naming the method `getPropertyMap()` might be clearer.  I don't think it's necessary to repeat the word "attachment" on a method of `IAttachment`.  And it's really an "attachment property" map, so I would expect "property" to be in the method name someplace.

Reply:
Another thing that would be _nice_ is if we could keep all this stuff separate from the attachments themselves.  It's really only necessary for game parsing, right?  It seems that this method is just cluttering the interface because it won't be used by anyone except `GameParser`.  The parser should be able to operate purely in terms of the public API of the attachments.